libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryService.java

```diff
@@ -253,6 +250,12 @@ private void checkApplicationContext() {
     }
   }
 
+  private void checkLocationEnabler() {
```

Comment:
As we're creating a new instance of `TelemetryLocationEnabler` if not already created, what about calling this method `createLocationEnabler` instead?